src/promptflow-core/promptflow/core/_prompty_utils.py

```diff
@@ -192,11 +193,11 @@ def get_open_ai_client_by_connection(connection, is_async=False):
     return client
 
 
-def send_request_to_llm(client, api, parameters):
+def send_request_to_llm(client, api, parameters, timeout):
     if api == "completion":
-        result = client.completions.create(**parameters)
+        result = client.with_options(timeout=timeout).completions.create(**parameters)
```

Comment:
Hi @ninghu , why the timeout needs to be configured with with_options? timeout can be also configured by create function passed by parameters.
![image](https://github.com/user-attachments/assets/8cddb609-295f-4d96-a55c-e1a24d0ff2f2)

Reply:
The change is by referencing the openai documentation here: https://github.com/openai/openai-python#timeouts

![image](https://github.com/user-attachments/assets/d5aa7db3-430f-478c-bd43-84a919188aa8)